File: android/app/src/main/java/com/phicdy/totoanticipation/model/scheduler/BootReceiver.java
package com.phicdy.totoanticipation.model.scheduler;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import com.phicdy.totoanticipation.model.storage.GameListStorage;
import com.phicdy.totoanticipation.model.storage.GameListStorageImpl;

import java.util.Date;

public class BootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.d("Boot", "receive boot");
        if (context == null || intent == null || intent.getAction() == null) return;
        if (intent.getAction().equals(Intent.ACTION_BOOT_COMPLETED)) {
            GameListStorage storage = new GameListStorageImpl(context);
            Date deadline = storage.totoDeadline();
            Log.d("Boot", "deadline: " + deadline.toString());
            long now = System.currentTimeMillis();
            if (deadline.getTime() == 0 || deadline.getTime() <= now) return;
            new DeadlineAlarm(context).setAtNoonOf(deadline);
        }
    }
}

File: android/legacy/src/main/java/com/phicdy/totoanticipation/legacy/model/scheduler/BootReceiver.kt
package com.phicdy.totoanticipation.legacy.model.scheduler

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import com.phicdy.totoanticipation.legacy.model.storage.GameListStorageImpl

class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context?, intent: Intent?) {
        Log.d("Boot", "receive boot")
        if (context == null || intent == null || intent.action == null) return
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            val storage = GameListStorageImpl(context)
            val deadline = storage.totoDeadline()
            Log.d("Boot", "deadline: $deadline")
            val now = System.currentTimeMillis()
            if (deadline.time == 0L || deadline.time <= now) return
            DeadlineAlarm(context).setAtNoonOf(deadline)
        }
    }
}
